### PR TITLE
Add ability to set arbitrary arguments in a renderer process

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This provides an easy way to send additional data to a renderer process preload script.  Something like `--my-app-channel=beta` for instance 😄

### CODE REVIEW

File: atom/browser/web_contents_preferences.cc [hunk 1]
```diff
@@ -136,6 +136,17 @@ void WebContentsPreferences::AppendExtraCommandLineSwitches(
...
+  base::ListValue* customArgs;
+  std::string customArg;
+  if (web_preferences.GetList(options::kCustomArgs, &customArgs)) {
+    for (size_t i = 0; i < customArgs->GetSize(); i++) {
+      if (customArgs->GetString(i, &customArg)) {
+        command_line->AppendArg(customArg);
+      }
+    }
+  }
+
```
Comment (ckerr) [inline atom/browser/web_contents_preferences.cc]: `DictionaryValue::GetList` is deprecated, and the newer API wraps around `std::vector<base::Value>` so I _think_ the nondeprecated version would look something like:

```c++
// Custom args for renderer process
const base::Value* customArgs = web_preferences.FindPath(options::kCustomArgs);
if ((customArgs != nullptr) && (customArgs->is_list())) {
  for (const base::Value& customArg : customArgs->GetList()) {
    if (customArg.is_string()) {
      command_line->AppendArg(customArg.GetString());
    }
  }
}
```

File: docs/api/browser-window.md [hunk 2]
```diff
@@ -353,6 +353,9 @@ It creates a new `BrowserWindow` with native properties as set by the `options`.
       script. You can use the `will-attach-webview` event on [webContents](web-contents.md)
       to strip away the `preload` script and to validate or alter the
       `<webview>`'s initial settings.
+    * `additionArguments` String[] (optional) - A list of strings that will be appended
+      to `process.argv` in the renderer process of this app.  Useful for passing smalls
```
Comment (ckerr) [inline docs/api/browser-window.md]: s/smalls/small/

File: spec/api-browser-window-spec.js [hunk 3]
```diff
@@ -1063,6 +1063,42 @@ describe('BrowserWindow module', () => {
...
+          show: false,
+          webPreferences: {
+            preload: preload,
+            additionalArguments: ['--my-magic-arg=foo']
+          },
+        })
+        w.loadURL(`file://${path.join(fixtures, 'api', 'blank.html')}`)
+      })
+    })
+
```
Comment (ckerr) [inline spec/api-browser-window-spec.js]: :+1:

**Comment by MarshallOfSound** [comment]
@ckerr Updated 👍

**Comment by codebytere** [comment]
@MarshallOfSound looks like this died on lint

**Pull request merged**